### PR TITLE
Remove application.secret and play.crypto.secret support

### LINKED ISSUE #8981: Remove support for `play.crypto.secret` & `application.secret` in 2.7.x
They have been marked as deprecated for a while 

`application.secret` since https://www.playframework.com/documentation/2.4.x/Migration24#Configuration-changes and `play.crypto.secret` since https://www.playframework.com/documentation/2.6.x/Migration26#Other-Configuration-changes

### PR DESCRIPTION
**Pull request by chipz**
## Fixes

Fixes #8981

## Purpose

Removed `play.crypto.secret` & `application.secret` leaving `play.http.secret.key` as the only way to set secret key

### CODE REVIEW

File: core/play/src/test/scala/play/api/http/SecretConfigurationParserSpec.scala [hunk 1]
```diff
@@ -14,25 +14,6 @@ class ActualKeySecretConfigurationParserSpec extends SecretConfigurationParserSp
...
-        ),
-        Environment.simple(mode = mode)
-      )
-      .secret
-      .secret
-  }
-
-}
-
 trait SecretConfigurationParserSpec extends Specification {
```
Comment (marcospereira) [inline core/play/src/test/scala/play/api/http/SecretConfigurationParserSpec.scala]: We can change this trait to be a class and use `play.http.secret.key` as the `secretKey` value. And then remove `ActualKeySecretConfigurationParserSpec` as well. :-)

File: core/play/src/main/scala/play/api/http/HttpConfiguration.scala [hunk 2]
```diff
@@ -285,7 +285,7 @@ object HttpConfiguration {
     val Blank = """\s*""".r
 
     val secret =
-      config.getDeprecated[Option[String]]("play.http.secret.key", "play.crypto.secret", "application.secret") match {
+      config.getDeprecated[Option[String]]("play.http.secret.key") match {
```
Comment (mkurz) [inline core/play/src/main/scala/play/api/http/HttpConfiguration.scala]: You can just use `get` here now instead of `getDeprecated`

Comment (chipz) [reply core/play/src/main/scala/play/api/http/HttpConfiguration.scala]: good point

File: core/play/src/test/scala/play/api/http/SecretConfigurationParserSpec.scala (hunk 1 above)
Comment (chipz) [reply core/play/src/test/scala/play/api/http/SecretConfigurationParserSpec.scala]: thanks for the review. done.

**Pull request merged**